model/pdata/generated_metrics_test.go

```diff
@@ -465,6 +465,32 @@ func TestMetric_Unit(t *testing.T) {
 	assert.EqualValues(t, testValUnit, ms.Unit())
 }
 
+func TestMetricDataType(t *testing.T) {
+	tv := NewMetric()
+	assert.Equal(t, MetricDataTypeNone, tv.DataType())
+	assert.Equal(t, "", MetricDataType(1000).String())
+	tv.SetDataType(MetricDataTypeGauge)
+	fillTestGauge(tv.Gauge())
+	assert.Equal(t, "Gauge", MetricDataTypeGauge.String())
+	assert.Equal(t, MetricDataTypeGauge, tv.DataType())
+	tv.SetDataType(MetricDataTypeSum)
+	fillTestSum(tv.Sum())
+	assert.Equal(t, "Sum", MetricDataTypeSum.String())
+	assert.Equal(t, MetricDataTypeSum, tv.DataType())
+	tv.SetDataType(MetricDataTypeHistogram)
+	fillTestHistogram(tv.Histogram())
+	assert.Equal(t, "Histogram", MetricDataTypeHistogram.String())
+	assert.Equal(t, MetricDataTypeHistogram, tv.DataType())
+	tv.SetDataType(MetricDataTypeExponentialHistogram)
+	fillTestExponentialHistogram(tv.ExponentialHistogram())
+	assert.Equal(t, "ExponentialHistogram", MetricDataTypeExponentialHistogram.String())
+	assert.Equal(t, MetricDataTypeExponentialHistogram, tv.DataType())
+	tv.SetDataType(MetricDataTypeSummary)
+	fillTestSummary(tv.Summary())
+	assert.Equal(t, "Summary", MetricDataTypeSummary.String())
+	assert.Equal(t, MetricDataTypeSummary, tv.DataType())
+}
```

Comment:
I feel like this test is "too much" for a unittest. You need to test the "SetDataType" and "DataType" APIs, and not the "enum" values for the "MetricDataType" enum.

Reply:
Makes sense. Actually I was thinking about generating enums definitions as well, but this would require more changes to the pdatagen logic. Let me know if you think it worth trying. I'll simplify this test anyway

Reply:
Let's simplify this test to do the "unit" testing. Then we can have a followup :)

Reply:
Done.

I accidentally pushed to a branch in this repo which I cannon update anymore. Submitted another PR instead https://github.com/open-telemetry/opentelemetry-collector/pull/4811